lightning/src/ln/channelmanager.rs

```diff
@@ -5958,6 +6002,23 @@ impl<Signer: Sign, M: Deref , T: Deref , K: Deref , F: Deref , L: Deref >
 				}
 			}
 		} else {
+			{
+				// First check if we can advance the channel type and try again.
+				let mut channel_state = self.channel_state.lock().unwrap();
+				if let Some(chan) = channel_state.by_id.get_mut(&msg.channel_id) {
+					if chan.get_counterparty_node_id() != *counterparty_node_id {
+						return;
+					}
+					if let Ok(msg) = chan.advance_channel_type_pref(self.genesis_hash) {
```

Comment:
There's some panics in `get_open_channel`. We're sure we'll never receive an `error_message` other than during the channel init process?